packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/s3/DownloadStaticESIndexService.java

```diff
@@ -0,0 +1,134 @@
+package software.uncharted.terarium.hmiserver.service.s3;
+
+import co.elastic.clients.elasticsearch.core.BulkResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import jakarta.annotation.PostConstruct;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.uncharted.terarium.hmiserver.models.mira.DKG;
+import software.uncharted.terarium.hmiserver.service.elasticsearch.ElasticsearchService;
+import software.uncharted.terarium.hmiserver.utils.JSONLInputStreamReader;
+
+/**
+ * Service to download static Elasticsearch index from S3 and initialize it.
+ */
+@Slf4j
+@Service
+@Configuration
+@RequiredArgsConstructor
+public class DownloadStaticESIndexService {
+
+	private final S3ClientService s3ClientService;
+	private final ElasticsearchService elasticsearchService;
+
+	@Value("${terarium.static-index-path}")
+	private String staticIndexPath;
+
+	private static final String EPI_ROOT = "epi_dkg_20241030";
```

Comment:
Should this be in the application properties for ease of update? And have potential different version based on environment?

Reply:
Yes it should!  However, this is happening during the `@PostConstruct` phase which apparently is before the configuration runs and grabs values from application.properties, so these all come in as `null`.